index.go

```diff
@@ -207,8 +210,13 @@ func (v *Index) RemoveAll(pathspecs []string, callback IndexMatchedPathCallback)
 
 //export indexMatchedPathCallback
 func indexMatchedPathCallback(cPath, cMatchedPathspec *C.char, payload unsafe.Pointer) int {
-	callback := (*IndexMatchedPathCallback)(payload)
-	return (*callback)(C.GoString(cPath), C.GoString(cMatchedPathspec))
+	if payload == nil {
+		return 0
+	} else if callback, ok := pointerHandles.Get(payload).(IndexMatchedPathCallback); ok {
+		return callback(C.GoString(cPath), C.GoString(cMatchedPathspec))
+	} else {
+		return -1
+	}
```

Comment:
It is optional to pass, but we only tell libgit2 to call `indexMatchedPathCallback` if the caller did pass in the callback. This function only gets called if there *is* a callback.